### PR TITLE
Refactor delly

### PR DESCRIPTION
**Pull request by johnmous**

### CODE REVIEW

**Comment by johnmous** [comment]
@ffinfo I see this test failure: Unable to delete directory /tmp/1495030236169-0/.log/test/config.
from nl.lumc.sasc.biopet.pipelines.flexiprep.FlexiprepTest.removeTempOutputDir(FlexiprepTest.scala:114)

and I dont see where it comes from (didnt touch anything in flexiprep)

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/svcallers/Delly.scala [hunk 1]
```diff
@@ -27,66 +27,80 @@ class Delly(val parent: Configurable) extends SvCaller {
...
-      if (tra) {
-        val delly = new DellyCaller(this)
+      if (bnd) {
+        val delly = new DellyCallerCall(this)
         delly.input = bamFile
-        delly.analysistype = "TRA"
+        delly.analysistype = "BND"
         delly.isIntermediate = true
-        delly.outputvcf = new File(dellyDir, sample + ".delly.tra.vcf")
+        delly.outputbcf = new File(dellyDir, sample + ".delly.tra.bcf")
```
Comment (ffinfo) [inline shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/svcallers/Delly.scala:90]: why bcf is used here? it's more use friendly to use vcf instead

File: shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala [hunk 2]
```diff
@@ -128,30 +129,39 @@ class ShivaSvCallingTest extends TestNGSuite with Matchers {
...
       pipeline.init()
       pipeline.script()
 
-      pipeline.functions.count(_.isInstanceOf[DellyCaller]) shouldBe
-        ((if (del) 1 else 0) + (if (dup) 1 else 0) + (if (inv) 1 else 0) + (if (tra) 1 else 0))
+      pipeline.functions.count(_.isInstanceOf[DellyCallerCall]) shouldBe
+        ((if (del) 1 else 0) + (if (dup) 1 else 0) + (if (inv) 1 else 0) + (if (bnd) 1 else 0) + (if (ins)
+                                                                                                    1
+                                                                                                  else
+                                                                                                    0))
```
Comment (ffinfo) [inline shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala]: This a bit ugly auto formatted. This has todo with the max line length. If you rewrite is like this:
``` scala
pipeline.functions.count(_.isInstanceOf[DellyCallerCall]) shouldBe
  ((if (del) 1 else 0) + (if (dup) 1 else 0) + (if (inv) 1 else 0) +
    (if (bnd) 1 else 0) + (if (ins)1 else 0))
```

This way the auto formater does need to use this nasty tricks ;)

File: shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala [hunk 3]
```diff
@@ -128,30 +129,39 @@ class ShivaSvCallingTest extends TestNGSuite with Matchers {
...
 
     val map = Map("sv_callers" -> List("delly"),
                   "delly" ->
-                    Map("DEL" -> del, "DUP" -> dup, "INV" -> inv, "TRA" -> tra))
+                    Map("DEL" -> del, "DUP" -> dup, "INV" -> inv, "BND" -> bnd, "INS" -> ins))
     val pipeline = initPipeline(map, outputDir)
 
     pipeline.inputBams = Map("bam" -> ShivaSvCallingTest.inputTouch("bam" + ".bam"))
 
-    if (!del && !dup && !inv && !tra) intercept[IllegalStateException] {
```
Comment (ffinfo) [inline shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala:146]: `tra` is not in there anymore?

File: biopet-extensions/src/main/scala/nl/lumc/sasc/biopet/extensions/delly/DellyCallerCall.scala [hunk 4]
```diff
@@ -29,33 +36,38 @@ class DellyCaller(val parent: Configurable) extends BiopetCommandLineFunction wi
   override def defaultCoreMemory = 4.0
 
   def versionCommand = versionexecutable.getAbsolutePath
-  def versionRegex = """DELLY \(Version: (.*)\)""".r
+  def versionRegex = """Delly \(Version: (.*)\)""".r
```
Comment (ffinfo) [inline biopet-extensions/src/main/scala/nl/lumc/sasc/biopet/extensions/delly/DellyCallerCall.scala]: Can this regex also work on both versions?

**Review by ffinfo — CHANGES_REQUESTED** [review]
Just a few things that need changes but in general it look fine. I still want to be sure the functional test does work on this.

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/svcallers/Delly.scala (hunk 1 above)
Comment (johnmous) [reply shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/svcallers/Delly.scala:90]: Unfortunately Delly only outputs bcf now

File: shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala (hunk 3 above)
Comment (johnmous) [reply shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala:146]: tra is now renamed to ```bnd```, for breakend

File: shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala (hunk 2 above)
Comment (johnmous) [reply shiva/src/test/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingTest.scala]: Formatted it as you suggested

File: biopet-extensions/src/main/scala/nl/lumc/sasc/biopet/extensions/delly/DellyCallerCall.scala (hunk 4 above)
Comment (johnmous) [reply biopet-extensions/src/main/scala/nl/lumc/sasc/biopet/extensions/delly/DellyCallerCall.scala]: Good point, fixed

**Comment by ffinfo** [comment]
Closing till we got reply from author why variants are missing

**Pull request closed without merging**